src/backends/pgfplotsx.jl

```diff
@@ -0,0 +1,95 @@
+using PGFPlotsX: PGFPlotsX
+
+const _pgfplotsx_linestyles = KW(
+    :solid => "solid",
+    :dash => "dashed",
+    :dot => "dotted",
+    :dashdot => "dashdotted",
+    :dashdotdot => "dashdotdotted",
+)
+
+const _pgfplotsx_markers = KW(
+    :none => "none",
+    :cross => "+",
+    :xcross => "x",
+    :+ => "+",
+    :x => "x",
+    :utriangle => "triangle*",
+    :dtriangle => "triangle*",
+    :circle => "*",
+    :rect => "square*",
+    :star5 => "star",
+    :star6 => "asterisk",
+    :diamond => "diamond*",
+    :pentagon => "pentagon*",
+    :hline => "-",
+    :vline => "|"
+)
+
+const _pgfplotsx_legend_pos = KW(
+    :bottomleft => "south west",
+    :bottomright => "south east",
+    :topright => "north east",
+    :topleft => "north west",
+    :outertopright => "outer north east",
+)
+# --------------------------------------------------------------------------------------
+# display calls this and then _display, its called 3 times for plot(1:5)
+function _update_plot_object(plt::Plot{PGFPlotsXBackend})
+    plt.o = PGFPlotsX.GroupPlot()
+
+    local axis
+    for sp in plt.subplots
+        bb = bbox(sp)
+        axis = PGFPlotsX.@pgf PGFPlotsX.Axis(
```

Comment:
I don't think you should use the `@pgf` macro at all. That macro is a way to be able to write things more conveniently but all it does is create a nested dictionary. But with it inside source code like this it feels pretty magic, while just using a `Dict` feels more familiar.

Reply:
https://kristofferc.github.io/PGFPlotsX.jl/stable/man/internals/#PGFPlotsX.Options

Reply:
Thanks for that hint!